NXP-30813: add CompoundDocument doctypes and facet

NourNuxeo opened the pull request.

bdelbosc commented on nuxeo-compound-documents/src/main/resources/META-INF/MANIFEST.MF:
```diff
@@ -3,3 +3,4 @@ Bundle-ManifestVersion: 2
 Bundle-Name: Nuxeo Compound Documents
 Bundle-SymbolicName: org.nuxeo.compound.documents;singleton:=true
 Bundle-Vendor: Nuxeo
+Nuxeo-Component: compound-documents-contrib.xml
```
OSGI-INF/compound ...

NourNuxeo commented:
Removed useless test dependency,
Deployed necessary components for minimal test
Renamed the compound bundle symbolic name to have the same `org.nuxeo.ecm` start as most nuxeo bundles do

kevinleturc commented on nuxeo-compound-documents/src/test/java/org/nuxeo/ecm/compound/documents/TestCompoundDocument.java:
```diff
@@ -0,0 +1,82 @@
+ * limitations under the License.
+ *
+ */
+package org.nuxeo.ecm.compound.documents;
```
I think we can get rid of `ecm` everywhere in this new package.

kevinleturc reviewed (commented):
I may be interesting to have automatic tests representing the acceptance criteria of the ticket.

troger commented on nuxeo-compound-documents/src/main/resources/OSGI-INF/compound-documents-contrib.xml:
```diff
@@ -0,0 +1,26 @@
+        <type>Picture</type>
+        <type>Video</type>
+        <type>Audio</type>
+        <type>CompoundDocument</type>
```
```suggestion
        <type>CompoundDocumentFolder</type>
```

troger commented on nuxeo-compound-documents/src/main/resources/OSGI-INF/compound-documents-contrib.xml:
```diff
@@ -0,0 +1,26 @@
+        <type>Picture</type>
+        <type>Video</type>
+        <type>Audio</type>
+        <type>CompoundDocument</type>
```
```suggestion
        <type>CompoundDocumentFolder</type>
```

troger commented on nuxeo-compound-documents/src/test/java/org/nuxeo/ecm/compound/documents/TestCompoundDocument.java:
```diff
@@ -0,0 +1,82 @@
+@RunWith(FeaturesRunner.class)
+@Features(CoreFeature.class)
+@RepositoryConfig(cleanup = Granularity.METHOD)
+@Deploy("org.nuxeo.ecm.core.management:OSGI-INF/core-management-framework.xml")
+@Deploy("org.nuxeo.ecm.core.management:OSGI-INF/core-management-types-contrib.xml")
```
Why are we requiring those contribs?

NourNuxeo replied:
got misled by ``` * service:org.nuxeo.runtime.stream.service.managment.contrib references missing [target=org.nuxeo.ecm.core.management.CoreManagementComponent;point=probes, target=org.nuxeo.ecm.core.management.CoreManagementComponent;point=healthCheck]``` when I was setting up deployments

troger commented on nuxeo-compound-documents/src/main/resources/OSGI-INF/compound-documents-contrib.xml:
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0"?>
+<component name="org.nuxeo.compound.doctype">
```
```suggestion
<component name="org.nuxeo.compound.documents.types">
```
(consistency with the group id)

troger commented on nuxeo-compound-documents/src/main/resources/META-INF/MANIFEST.MF:
```diff
@@ -3,3 +3,4 @@ Bundle-ManifestVersion: 2
 Bundle-Name: Nuxeo Compound Documents
 Bundle-SymbolicName: org.nuxeo.compound.documents;singleton:=true
 Bundle-Vendor: Nuxeo
+Nuxeo-Component: OSGI-INF/compound-documents-contrib.xml
```
I would have renamed the file to something more specific about what it does, such as `core-types-contrib.xml` (like in other packages).
WDYT @kevinleturc ?

troger commented on nuxeo-compound-documents/src/main/resources/OSGI-INF/core-types-contrib.xml:
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0"?>
+<component name="org.nuxeo.compound.documents.type">
```
```suggestion
<component name="org.nuxeo.compound.documents.types">
```
😬

NourNuxeo commented:
plural wording preference

troger reviewed (approved):
👍
